### ISSUE TITLE
Windows Debug Build Failure

### ISSUE THREAD

**Issue by HapeMask** [pytorch/pytorch#17617]
## 🐛 Bug

Building on Windows (VS2017 MSVC 15.9.2, toolset version 14.11, CUDA 10, CUDNN 7.4.1) with the official builder and `set DEBUG=1` fails with linker errors. It also appears that GCC-style args are being passed to `cl.exe` if debug mode is on. Release builds in the same configuration w/the same environment succeed.

## To Reproduce

1. Get latest builder scripts: `git clone https://github.com/pytorch/builder`
2. Setup env:
```
set VSDEVCMD_ARGS=-vcvars_ver=14.11
set PYTORCH_REPO=pytorch
set PYTORCH_BRANCH=8a21c6a5e
set BUILD_PYTHONLESS=1
set CUDA_VERSION=100
set CUDNN_VERSION=7
set "CONDA_LIB_PATH=C:/Users/{YOUR_USERNAME}/AppData/Local/Continuum/anaconda3/envs/{YOUR_CONDA_ENV_NAME}/Library/bin"
set DEBUG=1
```
3. Run build scripts:
```
cd builder\windows
cuda100.bat
```

The build fails with linker errors in `pool_op_cudnn.cc.obj`: https://gist.github.com/HapeMask/04cf175e6f4e43cd97cd05e23cfb2517

The failing linker command was:
```
FAILED: bin/caffe2_gpu.dll lib/caffe2_gpu.lib
cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_dll --intdir=caffe2\CMakeFiles\caffe2_gpu.dir --manifests  -- C:\PROGRA~2\MIB055~1\2017\PROFES~1\VC\Tools\MSVC\1411~1.255\bin\HostX64\x64\link.exe /nologo @CMakeFiles\caffe2_gpu.rsp  /out:bin\caffe2_gpu.dll /implib:lib\caffe2_gpu.lib /pdb:bin\caffe2_gpu.pdb /dll /version:0.0 /debug /INCREMENTAL:NO   && cd ."
LINK: command "C:\PROGRA~2\MIB055~1\2017\PROFES~1\VC\Tools\MSVC\1411~1.255\bin\HostX64\x64\link.exe /nologo @CMakeFiles\caffe2_gpu.rsp /out:bin\caffe2_gpu.dll /implib:lib\caffe2_gpu.lib /pdb:bin\caffe2_gpu.pdb /dll /version:0.0 /debug /INCREMENTAL:NO /MANIFEST /MANIFESTFILE:bin\caffe2_gpu.dll.manifest" failed (exit code 1120) with the following output:
   Creating library lib\caffe2_gpu.lib and object lib\caffe2_gpu.exp
```

There are also warnings which suggest that GCC-style args are being passed to `cl.exe`:
```[1473/1792] Building CXX object caffe2\CMakeFiles\math_test.dir\utils\math_test.cc.obj
cl : Command line warning D9002 : ignoring unknown option '-fno-omit-frame-pointer'
cl : Command line warning D9002 : ignoring unknown option '-O0'
```

## Environment

 - PyTorch Version (e.g., 1.0): 8a21c6a5e (also fails on master)
 - OS (e.g., Linux): Windows 10 v1809
 - How you installed PyTorch (`conda`, `pip`, source): source w/anaconda deps for building
 - Build command you used (if compiling from source): see above
 - Python version: 3.7
 - CUDA/cuDNN version: 10.0 / 7.4.1
 - GPU models and configuration: GTX 1080

**Comment by peterjc123** [comment]
Please apply this patch: https://github.com/pytorch/pytorch/pull/17201

**Comment by HapeMask** [comment]
Fixed in 2336f0ba062d3c00117ccefc49c07bf2626284f1, thanks!
